source/dub/init.d

```diff
@@ -90,6 +90,13 @@ void initPackage(Path root_path, string[string] deps, string type,
 private void initMinimalPackage(Path root_path, ref PackageRecipe p)
 {
 	p.description = "A minimal D application.";
+
+	// we want to create a empty, not-null dictionary
```

Comment:
I'd also mention the motivation here - looks suspicious for something that will be unclear when looking at it in a year or so, e.g. "so that the generated recipe contains an empty "dependencies" section in case of PackageFormat.json". Otherwise, LGTM.